File: JS&TS/leetcode/sorting/merge-intervals.js
/**
 * @param {number[][]} intervals
 * @return {number[][]}
 */
var merge = function (intervals) {
  const merged = [];

  const sortedIntervals = intervals.sort((a, b) => a[0] - b[0]);

  for (let interval of sortedIntervals) {
    if (merged.length && interval[0] <= merged.at(-1)[1]) {
      merged.at(-1)[1] = Math.max(interval[1], merged.at(-1)[1]);
    } else {
      merged.push(interval);
    }
  }
  return merged;
};

File: python/leetcode/sorting/merge-intervals.py
from typing import List


class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        merged = []

        for interval in sorted(intervals, key=lambda x: x[0]):
            if merged and interval[0] <= merged[-1][1]:
                merged[-1][1] = max(merged[-1][1], interval[1])
            else:
                merged.append(interval)
        return merged
